internal/api/handler.go

```diff
@@ -47,6 +52,55 @@ func (hdl *Handler) update(c *gin.Context) {
 	}
 }
 
+func (hdl *Handler) updateJSON(c *gin.Context) {
+
+	if c.Request.Header.Get("Content-Type") != "application/json" {
+		c.Status(http.StatusBadRequest)
+		return
+	}
+
+	var mtr Metrics
+	var metricValue string
+
+	if err := easyjson.UnmarshalFromReader(c.Request.Body, &mtr); err != nil {
+		c.Status(http.StatusBadRequest)
+		return
+	}
+
+	if mtr.Delta != nil {
+		metricValue = strconv.FormatInt(*mtr.Delta, 10)
+	}
+	if mtr.Value != nil {
+		metricValue = strconv.FormatFloat(*mtr.Value, 'f', -1, 64)
+	}
+	if mtr.Value == nil && mtr.Delta == nil {
+		c.Status(http.StatusBadRequest)
+		return
+	}

```

Comment:
Возможно стоит сделать проверку на `nil` первой, так как если она вернет `true` не смысла выполнять остальные?
```suggestion
	if mtr.Value == nil && mtr.Delta == nil {
		c.Status(http.StatusBadRequest)
		return
	}
	if mtr.Delta != nil {
		metricValue = strconv.FormatInt(*mtr.Delta, 10)
	}
	if mtr.Value != nil {
		metricValue = strconv.FormatFloat(*mtr.Value, 'f', -1, 64)
	}
```